app/services/course/assessment/submission/update_service.rb

```diff
@@ -12,7 +12,7 @@ def update
   def load_or_create_answers
     return unless @submission.attempting?
 
-    new_answers = questions_to_attempt.not_correctly_answered(@submission).attempt(@submission).
+    new_answers = questions_to_attempt.not_answered(@submission).attempt(@submission).
```

Comment:
It's for both, but worksheet is not affected now, because you cannot t submit answer half way in the worksheet now.

This won't create the new answer for an answered question unless user explicitly clicks the reattempt button.